packages/instantsearch.js/src/lib/utils/hydrateSearchClient.ts

```diff
@@ -85,9 +89,9 @@ export function hydrateSearchClient(
       requests: Object.keys(results).reduce(
         (acc, key) =>
           acc.concat(
-            results[key].rawResults.map((request) => ({
-              indexName: request.index,
-              params: request.params,
+            results[key].rawResults.map((result) => ({
+              indexName: result.index,
+              params: serializeQueryParameters(qs.parse(result.params)),
```

Comment:
it might or might not be right, but in v3 this is how the cache key is computed, also from the parameters, not from the result: 

https://github.com/algolia/algoliasearch-client-javascript/blob/5c1ce1308749aa0807940456393aa23ae46fb9eb/src/AlgoliaSearchCore.js#L232-L240